camel-support/src/main/java/io/kaoto/backend/camel/model/deployment/camelroute/IntegrationSpecSerializer.java

```diff
@@ -0,0 +1,39 @@
+package io.kaoto.backend.camel.model.deployment.camelroute;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.ser.std.StdSerializer;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+class IntegrationSpecSerializer extends StdSerializer<IntegrationSpec> {
+
+    public IntegrationSpecSerializer() {
+        super(IntegrationSpec.class);
+    }
+
+    @Override
+    public void serialize(IntegrationSpec data, JsonGenerator jsonGenerator, SerializerProvider serializerProvider)
+            throws IOException {
+
+        Map<String, Object> properties = new LinkedHashMap<>();
```

Comment:
Is it required to create this map ? 
I think it should be possible to use the ` JsonGenerator` instance directly and avoid generating temporary object ?

Reply:
I used the intermediate object to clean the nulls, but that's another option.

Reply:
can this be something configured at the serialization level ? like not including the null/empty objects ?  I'm fine with this at this point but we should do a second round where we try to improve all the object allocations

Reply:
The problem on this case comes from needing an  explicit different flows object because the `org.apache.camel.v1.integrationspec.Flow` class is too empty for us.

And that means having to override things and manually add the `_flows` as `flows` attribute while ignoring the other `flows` inherited attribute.

Reply:
I saw that but we should explore if we can use something like a `MixIn` or other similar features to customize of some specific types are customized. Or eventually wrapping the class all together.

Reply:
I removed the map and now use the jsonGenerator directly.

Should I try to look for a better solution on this PR?

Reply:
no it is fine